packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -59,7 +63,7 @@ export const MiddleTextTruncation = (props: MiddleTextTruncationProps) => {
       ref={ref}
       {...rest}
     >
-      {truncatedText}
+      {textRenderer ? textRenderer(truncatedText, text) : truncatedText}
```

Comment:
hm... we have this parent `span` element with some styles to make  it work.. wondering if passing some additional children which are going to wrap the truncated text could mess up something.. Not sure yet, but if your examples work, I guess its fine.

Reply:
Well, I tested it with the Select component, also tried it on a playground, and seems that it works fine even with nested component